dvc/repo/get.py

```diff
@@ -20,13 +21,27 @@ def __init__(self):
         )
 
 
+def resolve_artifact_addressing(path):  # add tests
+    parts = path.split(":")
+    if len(parts) == 1:
+        dvcyaml = None
+    elif len(parts) == 2:
+        dvcyaml, path = parts
+    else:
+        raise ValueError("more than one ':' found")  # todo replace error
+    if dvcyaml is not None and dvcyaml[-len(PROJECT_FILE) :] != PROJECT_FILE:
+        dvcyaml = os.path.join(dvcyaml, PROJECT_FILE)
+    return dvcyaml, path
```

Comment:
IMO A robust approach seems to be this: if ":" exists, split by the first occurrence of it (if there are more, will be part of the `path` since it's last

Reply:
We don't allow for `:` in artifact name, so it can't happen. Rather, if there is more that a single `:`, we can treat it as path (not an artifact ID). 🤔 if we should add it though.

Reply:
Posting what we discussed yesterday at refinement: it's fine to keep it like this. In future we'll collect feedback and make it work for other cases we can miss now.

Reply:
`:` is valid name for files.

Reply:
Since we removed `get` implementation from this PR, this is irrelevant here now. WE'll get back to it once we start implementing `get`.